modules/docs/arrow-docs/docs/docs/generic/coproduct/README.md

```diff
@@ -0,0 +1,167 @@
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.cop
+
+val apiResult = ServerError.cop<CommonServerError, RegistrationError, Registration>() //Returns Coproduct3<CommonServerError, RegistrationError, Registration>
+```
+
+All we have to do is provide the type parameters and we can make a Coproduct using the `cop` extension method. Just like `coproductOf`, if the type of the value isn't in the type parameters of the method call, it won't compile:
+
+```kotlin:ank
+//val apiResult = ServerError.cop<String, RegistrationError, Registration>()
+```
+
+##### fold
+
+Obviously, we're not just modeling errors for fun, we're going to handle them! All Coproducts have `fold` which allows us to condense the Coproduct down to a single type. For example, we could handle errors as such in a UI:
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.fold
+
+fun handleCommonError(commonError: CommonServerError) {
+}
+
+fun showCarAlreadyRegistered() {
+}
+
+fun callPolice() {
+}
+
+fun showCarSuccessfullyRegistered(car: Car) {
+}
+
+fun renderApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>) = apiResult.fold(
+            { commonError -> handleCommonError(commonError) },
+            { registrationError ->
+                when (registrationError) {
+                    CarAlreadyRegistered -> showCarAlreadyRegistered()
+                    StolenCar -> callPolice()
+                }
+            },
+            { registration ->
+                val registeredCar = registration.car
+
+                showCarSuccessfullyRegistered(registeredCar)
+            }
+    )
+```
+
+This example returns `Unit` because all of these are side effects, let's say our application was built for a command line and we just have to show a `String` for the result of the call (if only it was always that easy):
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.fold
+
+fun renderApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>): String = apiResult.fold(
+            { commonError ->
+                when (commonError) {
+                    ServerError -> "Server error, try again later."
+                    UserUnauthorized -> "Unauthorized!"
+                    OverRequestLimit -> "Too many api requests, try again later."
+                }
+            },
+            { registrationError ->
+                when (registrationError) {
+                    CarAlreadyRegistered -> "Car already registered."
+                    StolenCar -> "Car reported stolen!"
+                }
+            },
+            { registration ->
+                "Successfully Registered: $registration.car"
+            }
+    )
+```
+
+Here we're able to return the result of the `fold` and we're forced to handle all cases! Neat!
```

Comment:
Ah! There it is. Maybe just explicitly mention exhaustive evaluation here. You're saying it but still mentioning it by its real name should help.